x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -146,6 +181,172 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
                 return { createdAlerts: [], errors: {}, alertsWereTruncated: false };
               }
             },
+            alertWithSuppression: async (
```

Comment:
Does this function need to handle `maxAlerts` as well?

Reply:
I didn't have a need for it since the deduplication strategy for `alertWithSuppression` is different from `alertWithPersistence`. In `alertWithPersistence`, I wanted to apply maxSignals after deduplication to avoid situations where maxSignals gets filled by many duplicate alerts. For `alertWithSuppression`, the rule logic is responsible for deduplicating and `alertWithSuppression` assumes that every alert passed in will either update an existing alert or create a new alert. 

Since we aggregate source docs and provide only the count and time range bounds for the source docs, `alertWithSuppression` would not be able to determine which source docs were dupes. So the rule logic explicitly excludes time ranges that it has found docs in in previous rule executions. `alertWithPersistence` receives alert docs that correspond 1-1 with source docs, so it can determine if an alert has already been generated for a particular source doc.

If we decide that updated alerts in the future should not count towards masSignals, for example, then I think we'd want to add a maxSignals param to `alertWithSuppression`.

Reply:
Gotcha, that makes sense, thanks for confirming!